infracost-usage-example.yml

```diff
@@ -272,15 +272,70 @@ resource_usage:
+      monthly_bulk_select_data_returned_gb: 500000      # Monthly data returned by S3 Select in GB (for bulk level of S3 Glacier)
+      monthly_expedited_select_data_scanned_gb: 500000  # Monthly data scanned by S3 Select in GB (for expedited level of S3 Glacier)
+      monthly_expedited_select_data_returned_gb: 500000 # Monthly data returned by S3 Select in GB (for expedited level of S3 Glacier)
+      monthly_standard_data_retrieval_requests: 500000  # Monthly data Retrieval requests (for standard level of S3 Glacier).
+      monthly_expedited_data_retrieval_requests: 500000 # Monthly data Retrieval requests (for expedited level of S3 Glacier).
+      monthly_standard_data_retrieval_gb: 5000          # Monthly data retrievals in GB (for standard level of S3 Glacier).
+      monthly_expedited_data_retrieval_gb: 5000         # Monthly data retrievals in GB (for expedited level of S3 Glacier).
+      early_delete_gb: 500000                           # If an archive is deleted within 3 months of being uploaded, you will be charged an early deletion fee per GB.
+    glacier_deep_archive: # Usages of S3 Glacier Deep Archive:
+      storage_gb: 60000                                # Total storage in GB.
+      monthly_tier_1_requests: 6000000                 # Monthly PUT, COPY, POST, LIST requests (Tier 1).
+      monthly_tier_2_requests: 600000                  # Monthly GET, SELECT, and all other requests (Tier 2).
+      monthly_lifecycle_transition_requests: 600000    # Monthly Lifecycle Transition requests.
+      monthly_standard_data_retrieval_requests: 600000 # Monthly data Retrieval requests (for standard level of S3 Glacier).
+      monthly_bulk_data_retrieval_requests: 600000     # Monthly data Retrieval requests (for bulk level of S3 Glacier).
+      monthly_standard_data_retrieval_gb: 6000         # Monthly data retrievals in GB (for standard level of S3 Glacier).
+      monthly_bulk_data_retrieval_gb: 6000             # Monthly data retrievals in GB (for bulk level of S3 Glacier).
+      early_delete_gb: 600000                          # If an archive is deleted within 6 months of being uploaded, you will be charged an early deletion fee per GB.
+
+  aws_s3_bucket.my_bucket:
+    object_tags: 10000000 # Total object tags. Only for AWS provider V3.
     standard: # Usages of S3 Standard:
       storage_gb: 10000 # Total storage in GB.
       monthly_tier_1_requests: 1000000 # Monthly PUT, COPY, POST, LIST requests (Tier 1).
       monthly_tier_2_requests: 100000 # Monthly GET, SELECT, and all other requests (Tier 2).
       monthly_select_data_scanned_gb: 10000 # Monthly data scanned by S3 Select in GB.
       monthly_select_data_returned_gb: 1000 # Monthly data returned by S3 Select in GB.
-    intelligent_tiering: # Usages of S3 Intelligent - Tiering:
+    intelligent_tiering: # Usages of S3 Intelligent - Tiering (Only for AWS provider V3):
```

Comment:
I think you should still be able to model different storage classes using the usage file for v4. We shouldn't show the prices by default but if a user is using that storage class with the bucket and they add usage, we should show it since you don't need top use lifecycle policies to use different storage classes.

Reply:
Do I understand correctly the logic?
1. If there are rules on the TF resource - it's v3 and we need to use them for cost components.
2. If there are no any rules, but the usage is present - it's _maybe_ v4 and we can use usage params to show the cost components? <- I see an issue for v3 here when there are no rules, but there are usage params.

Reply:
Sounds good, I'll update the PR. Thanks! 👍

Reply:
I misunderstood how the code works, it's already exactly how you described 😅. Thank you! Updated the tests in https://github.com/infracost/infracost/pull/1370/commits/3557f33f7fea344c6b291568c574898de87e8c8b.